studio/pages/authorize.tsx

```diff
@@ -0,0 +1,232 @@
+
+const APIAuthorizationPage: NextPageWithLayout = () => {
+  const { ui } = useStore()
+  const router = useRouter()
+  const { auth_id } = useParams()
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const [selectedOrg, setSelectedOrg] = useState<string>()
+
+  const { data: organizations, isLoading: isLoadingOrganizations } = useOrganizationsQuery()
+  const { data: requester, isLoading, isError, error } = useApiAuthorizationQuery({ id: auth_id })
+  const isApproved = requester?.approved_at !== null
+  const isExpired = dayjs().isAfter(dayjs(requester?.expires_at))
+
+  const { mutateAsync: approveRequest } = useApiAuthorizationApproveMutation()
+  const { mutateAsync: declineRequest } = useApiAuthorizationDeclineMutation()
+
+  useEffect(() => {
+    if (!isLoadingOrganizations) {
+      setSelectedOrg(organizations?.[0].slug ?? undefined)
+    }
+  }, [isLoadingOrganizations])
+
+  const onApproveRequest = async () => {
+    if (!auth_id) {
+      return ui.setNotification({
+        category: 'error',
+        message: 'Unable to approve request: auth_id is missing ',
+      })
+    }
+    if (!selectedOrg) {
+      return ui.setNotification({
+        category: 'error',
+        message: 'Unable to approve request: No organization selected',
+      })
+    }
+
+    try {
+      setIsSubmitting(true)
+      const res = await approveRequest({ id: auth_id, organization_id: selectedOrg })
+      router.push(res.url)
+    } catch (error: any) {
+      ui.setNotification({
+        category: 'error',
+        message: `Failed to approve request: ${error.message}`,
+      })
+      setIsSubmitting(false)
+    }
+  }
+
+  const onDeclineRequest = async () => {
+    if (!auth_id)
+      return ui.setNotification({
+        category: 'error',
+        message: 'Unable to decline request: auth_id is missing ',
+      })
+
+    try {
+      setIsSubmitting(true)
+      await declineRequest({ id: auth_id })
+      ui.setNotification({ category: 'success', message: 'Declined API authorization request' })
+      router.push('/projects')
+    } catch (error: any) {
+      ui.setNotification({
+        category: 'error',
+        message: `Failed to decline request: ${error.message}`,
+      })
+      setIsSubmitting(false)
+    }
+  }
+
+  if (auth_id === undefined) {
+    return (
+      <FormPanel header={<p>Authorization for API access</p>}>
+        <div className="w-[500px] px-8 py-6">
+          <Alert withIcon variant="warning" title="Missing authorization ID">
+            Please provide a valid authorization ID in the URL
+          </Alert>
+        </div>
+      </FormPanel>
+    )
+  }
+
+  if (isLoading) {
+    return (
+      <FormPanel header={<p>Authorize API access</p>}>
+        <div className="w-[500px] px-8 py-6 space-y-2">
+          <ShimmeringLoader />
+          <ShimmeringLoader className="w-3/4" />
+          <ShimmeringLoader className="w-1/2" />
+        </div>
+      </FormPanel>
+    )
+  }
```

Comment:
Should the order of these two be swapped? So the loading state shows while `auth_id` is `undefined` initially thanks to Next

Reply:
oh yeah! ahh i forgot to push the changes up good catch 🙏